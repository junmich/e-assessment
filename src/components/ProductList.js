import React from 'react';
import List from './List';

const ProductList = ({ products, selectProduct, deleteProduct }) => {
    return (
        <div className="product-list">
            <div className="product-list-title">Products</div>
            <div>
                {products.map((product, key) => <List key={key} product={product} handleSelectProduct={() => selectProduct(product)} handleDeleteProduct={() => deleteProduct(product)} />)}
            </div>
        </div>
        
    );
}

ProductList.defaultProps = {
    products: []
};

export default ProductList;