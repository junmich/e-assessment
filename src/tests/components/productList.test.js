import React from 'react';
import { render } from 'enzyme';
import ProductList from '../../components/ProductList';

describe('Product List Test', () => {
    const selectProduct = jest.fn();
    const deleteProduct = jest.fn();
    const products = [{ name: 'test', description: 'test' }];
    test('should render product list correctly', () => {
        const wrapper = render(<ProductList products={products} deleteProduct={deleteProduct} selectProduct={selectProduct} />);
        expect(wrapper.find('.list').length).toBe(1);
        expect(wrapper.find('.product-list-title').length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render product list correctly with no product', () => {
        const wrapper = render(<ProductList products={[]} deleteProduct={deleteProduct} selectProduct={selectProduct} />);
        expect(wrapper.find('.list').length).toBe(0);
        expect(wrapper.find('.product-list-title').length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });
});

