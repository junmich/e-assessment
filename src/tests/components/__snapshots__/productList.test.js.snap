// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product List Test should render product list correctly 1`] = `
<div
  class="product-list"
>
  <div
    class="product-list-title"
  >
    Products
  </div>
  <div>
    <div
      class="list"
    >
      <div
        class="list__text"
      >
        test
        <div
          class="list__subtitle"
        >
          test
        </div>
      </div>
      <div>
        <button
          class="default-button"
        >
          Edit
        </button>
        <button
          class="delete-button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Product List Test should render product list correctly with no product 1`] = `
<div
  class="product-list"
>
  <div
    class="product-list-title"
  >
    Products
  </div>
  <div>
    
  </div>
</div>
`;
